src/Microsoft.ML.Data/Training/TrainerUtils.cs

```diff
@@ -382,6 +383,24 @@ public static SchemaShape.Column MakeR4ScalarWeightColumn(string weightColumn)
                 return null;
             return new SchemaShape.Column(weightColumn, SchemaShape.Column.VectorKind.Scalar, NumberType.R4, false);
         }
+
+        /// <summary>
+        /// Check that the label, feature, weights is not supplied in the args the constructor.
+        /// Those parameters should be internal if they are not used from the maml help code path.
+        /// </summary>
+        public static void CheckArgumesDefaultColNames(string defaultColName, string argValue)
+        {
+            if (argValue != defaultColName)
+                throw Contracts.Except($"Don't supply a value for the {defaultColName} column in the arguments, as it will be ignored. Specify them in the loader, or constructor instead instead.");
+        }
+
+        public static void CheckArgsAndAdvancedSettingMismatch<T>(T methodParam, T defaultVal, T setting, string argName)
+        {
+            // if, after applying the advancedArgs delegate, the args are different that the default value
+            // and are also different than the value supplied directly to the xtension method, warn the user.
+            if (!setting.Equals(defaultVal) && !setting.Equals(methodParam))
+                Console.WriteLine($"The value supplied to advanced settings , is different than the value supplied directly. Using value {setting} for {argName}");
```

Comment:
> Console.WriteLine [](start = 15, length = 18)

I don't like this. Can we accept Host or Channel as first object and use it?

Reply:
for sure; put it to look for a Warn method, or pass the channel, and forgot to replace. 

---
In reply to: [220018209](https://github.com/dotnet/machinelearning/pull/1009#discussion_r220018209) [](ancestors = 220018209)